pkg/authorization/toplevel_org_authorizer.go

```diff
@@ -100,7 +100,7 @@ func (a *topLevelOrgAccessAuthorizer) Authorize(ctx context.Context, attr author
 	if subjectCluster := attr.GetUser().GetExtra()[authserviceaccount.ClusterNameKey]; len(subjectCluster) > 0 {
 		// service account will automatically get access to its top-level org
 		subjectTopLevelOrg, ok := topLevelOrg(logicalcluster.New(subjectCluster[0]))
```

Comment:
If you're changing requestTopLevelOrg, should you also change subjectTopLevelOrg?